### DISCUSSION TITLE
Cannot start llama.cpp on Windows, no output

### DISCUSSION THREAD

**Discussion by waltersamson** [ggml-org/llama.cpp#10926] [Q&A]
My setup:
- Multiple AVX2 Intel and AMD Windows boxes
- llama-b4372-bin-win-avx2-x64
- Windows 10

I feel stupid now. I've been using llama.cpp on Linux for quite some time without issue. Today I tried running it on Windows, somehow I ran into a strange problem:

- Download llama-b4372-bin-win-avx2-x64.zip from Releases
- Execute llama-cli.exe or llama-server.exe

Expected:
The executable should print something, such as "build: 4337 (160bc039) with cc (Ubuntu 13.2.0-23ubuntu4) 13.2.0 for x86_64-linux-gnu"

However, in my case, the program just returns, with no output whatsoever, it looks like this:
```
F:\llm\apps\llama.cpp\llama-b4372-bin-win-avx2-x64>llama-server.exe

F:\llm\apps\llama.cpp\llama-b4372-bin-win-avx2-x64>llama-cli.exe

F:\llm\apps\llama.cpp\llama-b4372-bin-win-avx2-x64>llama-gguf.exe
usage: llama-gguf.exe data.gguf r|w [n]
r: read data.gguf file
w: write data.gguf file
n: no check of tensor data

F:\llm\apps\llama.cpp\llama-b4372-bin-win-avx2-x64>
```

The binary llama-gguf.exe did run, but not llama-server.exe and llama-cli.exe.

Then I tried giving it a complete cmdline, but still nothing happened:
```
F:\llm\apps\llama.cpp\llama-b4372-bin-win-avx2-x64>llama-cli.exe -m ..\..\..\models\volatile\Qwen2.5-14B-Instruct-Q8_0.gguf -co -cnv -p "You are a helpful assistant." -fa -n 512 -c 8192

F:\llm\apps\llama.cpp\llama-b4372-bin-win-avx2-x64>
```

I feel stupid now, I must be missing an obvious piece.

Thanks for any help!

**Comment by Martin-HZK** [comment]
Same question, have you figure out why? Everything works fine on windows but not windows

**Reply by waltersamson** [reply]
> Same question, have you figure out why? Everything works fine on windows but not windows

Unfortunately no, I've opened an issue: https://github.com/ggerganov/llama.cpp/issues/10944

**Comment by 8forty** [comment]
I've had this problem pop up several times, and the only clue I've found so far is that this:
```
dev\github\llama.cpp\build\bin\Release\llama-server --help
```
fails (no error, no output), but then this:
```
cd dev
github\llama.cpp\build\bin\Release\llama-server --help
```
works (i.e. prints the help stuff)!  And it continues to work from each of the subdirs down to Release, so it seems there's a problem with the longer path for llama-server (but not most(?) of the other exe's).

Note that when --help doesn't work, llama-server also fails/does nothing regardless of which options I use.

**Reply by slaren** [reply]
Does this happen with the latest version? Did you build it yourself or are you using the releases?

**Reply by 8forty** [reply]
Yeah I did a fresh clone, and built it with msys.  The latest releases work fine, so I guess my builds have something different/missing vs. the release builds.

**Reply by slaren** [reply]
Can you run it under a debugger to see where it crashes?

**Reply by 8forty** [reply]
I tried building/debugging with visual studio 2022 community, but too many errors and I don't have enough time now to pursue it further.

**Comment by electroficator** [comment]
I made two actions:
1) Update MSVC 2015-2022 runtime (to The latest version is v14.44.35211.0)
2) add path to llama to PATH environment variable then restart.
llama works!

**Reply by felladrin** [reply]
Updating MSVC is probably the answer OP was looking for.
Additional info: Link for downloading the latest MSVC redistributable can be found at https://learn.microsoft.com/en-us/cpp/windows/latest-supported-vc-redist?view=msvc-170#latest-supported-redistributable-version
